packages/account/src/BiconomySmartAccount.ts

```diff
@@ -284,12 +284,49 @@ export class BiconomySmartAccount extends SmartAccount implements IBiconomySmart
     return userOp
   }
 
+  private validateUserOpAndRequest(
+    userOp: Partial<UserOperation>,
+    tokenPaymasterRequest: BiconomyTokenPaymasterRequest
+  ): void {
+    if (!userOp.callData) {
+      throw new Error('Userop callData cannot be undefined')
+    }
+
+    const feeTokenAddress = tokenPaymasterRequest?.feeQuote?.tokenAddress
+    Logger.log('requested fee token is ', feeTokenAddress)
+
+    if (!feeTokenAddress || feeTokenAddress == ethers.constants.AddressZero) {
+      throw new Error(
+        'Invalid or missing token address. Token address must be part of the feeQuote in tokenPaymasterRequest'
+      )
+    }
+
+    const spender = tokenPaymasterRequest?.spender
+    Logger.log('fee token approval to be checked and added for spender: ', spender)
+
+    if (!spender || spender == ethers.constants.AddressZero) {
+      throw new Error(
+        'Invalid or missing spender address. Sepnder address must be part of tokenPaymasterRequest'
+      )
+    }
+  }
+
   /** Optional method to update the userOp.calldata with batched transaction which approves the paymaster spender(if required)
    * with necessary amount*/
```

Comment:
Remove this comment. It's already included in below method comment under @notice section.

Reply:
done